### PR TITLE
Feature - Allow participants to register

### PR DESCRIPTION
**Pull request by v-stamenova**
For testing - migrate:fresh, db:seed:
- [x] As a content mod you first have to create slots and schedule the presentations so that they become available to the user
- [x] Register as a user and go to my programme and from there go to register for presentation
- [x] The already full presentations will be marked in gray
- [x] The presentations that are available are in purple - enroll by clicking on them
- [x] The presentations that you are enrolled in are in dark gray - disenroll by clicking on them
- [x] There are some validations in place - for enrolling: you cannot enroll in a presentation you are already enrolled in; you cannot enroll in a presentation if during the same time you have another presentation; you cannot enroll if the max participants is reached; for disenrolling: you cannot disenroll in a presentation you are not enrolled in

The design is questionable to say the least, feel free to fix it and I'm waiting for the PR with the view of the presentation so that we make the clicking of a block to just show the presentation and the enrolling/disenrolling to happen from that page

### CODE REVIEW

**Comment by v-stamenova** [comment]
Bug: When speakers enrol for presentations it is not checked if they are not leading a lecture at that moment. Don't merge yet!

File: app/Http/Controllers/HubController.php [hunk 1]
```diff
@@ -2,46 +2,101 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Presentation;
+use App\Models\Timeslot;
 use App\Models\User;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\DB;
```
Comment (TimKardol) [inline app/Http/Controllers/HubController.php:11]: I think @dwaard mentioned something about not using this if possible?

File: app/Http/Controllers/HubController.php [hunk 2]
```diff
@@ -2,46 +2,101 @@
...
+
+            if ($presentationStart >= $enrolledEnd) {
+                continue;
+            }
+
+            return redirect()->back()
+                ->banner('You have enrolled already in a presentation that takes place at the same time as this one');
+        }
+
+        DB::transaction(function () use ($presentation) {
```
Comment (TimKardol) [inline app/Http/Controllers/HubController.php:83]: I see the DB used here. I'd like @dwaard to take a look.

File: database/seeders/ParticipantSeeder.php [hunk 3]
```diff
@@ -5,6 +5,8 @@
 use App\Models\Presentation;
 use App\Models\Speaker;
 use App\Models\User;
+use Database\Factories\PresentationFactory;
+use Database\Factories\UserFactory;
 use Illuminate\Database\Console\Seeds\WithoutModelEvents;
 use Illuminate\Database\Seeder;
 use Illuminate\Support\Facades\DB;
```
Comment (TimKardol) [inline database/seeders/ParticipantSeeder.php:12]: Same thing here.

File: routes/web.php [hunk 4]
```diff
@@ -41,14 +41,20 @@
     //route for announcements
     Route::get('/dashboard/announcements', [HubController::class, 'getAnnouncements'])->name('announcements');
 
-    //route for my profile in personal hun
+    //route for my profile in personal hub
     Route::get('/dashboard/profile', [HubController::class, 'getProfileInfo'])->name('my-profile');
 
     //route for personal programme
-    Route::get('/dashboard/programme', [HubController::class, 'getProgramme'])->name('my-programme');
+    Route::get('/dashboard/programme', [HubController::class, 'personalProgramme'])->name('my-programme');
```
Comment (TimKardol) [inline routes/web.php:48]: When this gets merged: I have renamed the routes to 'myconference/...' in PR #149 . Keep that in mind.

**Review by TimKardol — CHANGES_REQUESTED** [review]
Feedback:

- As we talked about the UI/UX needs a complete rework. I think that is already on the backlog. 
- There is no clear validation that someone enrolled. So a confirmation popup or something else should be added so that people know they enrolled for a lecture/workshop. 
- Colors: this is part of the UI/UX rework. Colors right now do not follow standards so it is confusing on what is actually happening.

**Pull request closed without merging**